pcap/pcapio/read.go

```diff
@@ -119,47 +119,46 @@ func (r *Reader) readHeader() error {
 	return nil
 }
 
-func (r *Reader) TsFromHeader(hdr []byte) nano.Ts {
+func (r *PcapReader) TsFromHeader(hdr []byte) nano.Ts {
 	ns := int64(r.byteOrder.Uint32(hdr[0:4])) * 1_000_000_000
 	ns += int64(r.byteOrder.Uint32(hdr[4:8]) * r.nanoSecsFactor)
 	return nano.Ts(ns)
 }
 
-func (r *Reader) ReadBlock(span nano.Span) ([]byte, error) {
+func (r *PcapReader) Read() ([]byte, Info, error) {
 	header := r.header
 	if header != nil {
 		r.header = nil
-		return header, nil
+		return header, Info{Type: TypeSection}, nil
 	}
-again:
 	hdr, err := r.Reader.Peek(packetHeaderLen)
 	if err != nil {
-		return nil, err
+		if err == io.EOF {
+			err = nil
+		}
+		return nil, Info{}, err
 	}
 	caplen := int(r.byteOrder.Uint32(hdr[8:12]))
 	fullLength := int(r.byteOrder.Uint32(hdr[12:16]))
 	if caplen > int(r.snaplen) {
-		return nil, fmt.Errorf("capture length exceeds snap length: %d > %d", caplen, r.snaplen)
+		return nil, Info{}, fmt.Errorf("capture length exceeds snap length: %d > %d", caplen, r.snaplen)
 	}
 	if caplen > fullLength {
-		return nil, fmt.Errorf("capture length exceeds original packet length: %d > %d", caplen, fullLength)
+		return nil, Info{}, fmt.Errorf("capture length exceeds original packet length: %d > %d", caplen, fullLength)
 	}
 	// pull out timestamp before reading rest of packet as header can get clobbered
 	ts := r.TsFromHeader(hdr)
 	pkt, err := r.Reader.Read(caplen + packetHeaderLen)
 	if err != nil {
-		return nil, err
-	}
-	r.Offset += uint64(caplen + packetHeaderLen)
-	if !span.ContainsClosed(ts) {
-		goto again
+		return nil, Info{}, err
 	}
-	return pkt, nil
+	r.offset += uint64(caplen + packetHeaderLen)
+	return pkt, Info{Type: TypePacket, Link: r.linkType, Ts: ts}, nil
 }
 
 // ReadPacketData reads next packet from file.
```

Comment:
```suggestion
// readPacketData reads next packet from file.
```

Reply:
Actually, this code is  unused and subsumed by Read() so I will delete it.